www_src/pages/section-2/grid.jsx

```diff
@@ -0,0 +1,382 @@
+var React = require('react');
+
+var Slot = require('./slot.jsx');
+var Page = require('./page.jsx');
+var AddPage = require('./add-page.jsx');
+
+module.exports = React.createClass({
+  /**
+   * Zoom in to a page tile
+   * @param  {Number} pageX Page's x coordinate in grid
+   * @param  {Number} pageY Page's y coordinate in grid
+   * @param  {Number} pagesWide Number of pages to fit in width-wise
+   * @param  {boolean} doCentering Determine whether or not to move "camera" to center the target
+   */
+  calculateCameraState: function (pageX, pageY, pagesWide, doCentering) {
```

Comment:
hmm, this feels like a slug of code that belongs in its own plain js library, generating the new data based on the current input